src/pages/index.js

```diff
@@ -3,82 +3,296 @@ import FormValidator from "../components/FormValidator.js";
+/* -------------------------------------------------------------------------- */
+
+getCards();
+
+let cardSection;
+
+function getCards() {
+  api
+    .getInitialCards()
+    .then((cards) => {
+      cardSection = new Section(
+        {
+          items: cards,
+          renderer: (card) => {
+            const cardElement = createCard(card);
+            cardSection.addItem(cardElement);
+          },
+        },
+        "#el-card-list"
+      );
+      cardSection.renderCards(cards);
+    })
+    .catch((error) => {
+      console.error("Failed to fetch cards:", error);
+    });
+}
+
+/* -------------------------------------------------------------------------------------------- */
+/*      Create new card and store the info in the server, then call to render card object       */
+/* -------------------------------------------------------------------------------------------- */
+
+function createNewCard(card) {
+  variables.createCardButton.textContent = "Creating...";
+  api
+    .createCard(card)
+    .then((cardData) => {
+      cardSection.addItem(createCard(cardData));
+      addCardPopup.close();
+      variables.addModalForm.reset();
+    })
+    .catch((error) => {
+      console.error("Failed to create new card", error);
+    })
+    .finally(() => {
+      variables.createCardButton.textContent = "Create";
+    });
+}
+
+/* -------------------------------------------------------------------------- */
+/*               Call API to submit card info to be created                   */
+/* -------------------------------------------------------------------------- */
+
+const addCardPopup = new PopupWithForm(
   "#profile-add-modal",
   ({ title: name, link }) => {
-    renderCard({ name, link });
-    variables.addModalForm.reset();
+    createNewCard({ name, link });
     addModalFormValidator.disableSubmitButton();
   }
 );
 
-addFormPopup.setEventListeners();
+/* -------------------------------------------------------------------------- */
+/*                    Add Card Form Event Listener                            */
+/* -------------------------------------------------------------------------- */
+
+addCardPopup.setEventListeners();
 variables.addButton.addEventListener("click", () => {
-  addFormPopup.open();
+  addCardPopup.open();
 });
-//
-const cardSection = new Section(
-  {
-    items: data,
-    renderer: renderCard,
-  },
-  "#el-card-list"
+
+/* -------------------------------------------------------------------------- */
+/*                         Call API and pass info to update the avatar        */
+/* -------------------------------------------------------------------------- */
+
+const avatarPopupForm = new PopupWithForm(
+  "#edit-avatar",
+  ({ link: avatar }) => {
+    variables.avatarModalSaveButton.textContent = "Saving...";
+    api
+      .updateUserAvatar({ avatar })
+      .then((res) => {
+        variables.avatarIcon.src = res.avatar;
```

Comment:
Class `UserInfo` should set the entire user’s data (including the **avatar**)

Reply:
Hi Gennady, could you please elaborate on this comment. My understanding is that the class does take all three inputs: name, description, and avatar?

Reply:
> Hi Gennady, could you please elaborate on this comment. My understanding is that the class does take all three inputs: name, description, and avatar?


You've done everything right. Now you have method `updateAvatar` for that